hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/volume/TestPeriodicVolumeChecker.java

```diff
@@ -79,40 +81,54 @@ public void testPeriodicVolumeChecker() throws Exception {
 
     DatanodeConfiguration dnConf =
         conf.getObject(DatanodeConfiguration.class);
-    dnConf.setDiskCheckMinGap(Duration.ofMinutes(2));
-    dnConf.setPeriodicDiskCheckIntervalMinutes(1);
-    conf.setFromObject(dnConf);
-
-    DatanodeDetails datanodeDetails =
-        ContainerTestUtils.createDatanodeDetails();
-    OzoneContainer ozoneContainer =
-        ContainerTestUtils.getOzoneContainer(datanodeDetails, conf);
-    MutableVolumeSet dataVolumeSet = ozoneContainer.getVolumeSet();
-
-    StorageVolumeChecker volumeChecker = dataVolumeSet.getVolumeChecker();
-    volumeChecker.setDelegateChecker(
-        new TestStorageVolumeChecker.DummyChecker());
-
-    // 1 for volumeSet and 1 for metadataVolumeSet
-    // in MutableVolumeSet constructor
-    Assert.assertEquals(2, volumeChecker.getNumAllVolumeChecks());
-    Assert.assertEquals(0, volumeChecker.getNumAllVolumeSetsChecks());
-
-    // wait for periodic disk checker start
-    Thread.sleep((60 + 5) * 1000);
-
-    // first round
-    // 2 for volumeSet and 2 for metadataVolumeSet
-    Assert.assertEquals(4, volumeChecker.getNumAllVolumeChecks());
-    Assert.assertEquals(1, volumeChecker.getNumAllVolumeSetsChecks());
-    Assert.assertEquals(0, volumeChecker.getNumSkippedChecks());
-
-    // wait for periodic disk checker next round
-    Thread.sleep((60 + 5) * 1000);
-
-    // skipped next round
-    Assert.assertEquals(4, volumeChecker.getNumAllVolumeChecks());
-    Assert.assertEquals(1, volumeChecker.getNumAllVolumeSetsChecks());
-    Assert.assertEquals(1, volumeChecker.getNumSkippedChecks());
+    Duration gap = dnConf.getDiskCheckMinGap();
+    Duration interval = Duration.ofMinutes(
+        dnConf.getPeriodicDiskCheckIntervalMinutes());
+
+    FakeTimer timer = new FakeTimer();
+
+    StorageVolumeChecker volumeChecker = new StorageVolumeChecker(conf, timer);
+
+    try {
+      volumeChecker.registerVolumeSet(new ImmutableVolumeSet(makeVolumes(
+          2, HEALTHY)));
+      volumeChecker.registerVolumeSet(new ImmutableVolumeSet(makeVolumes(
+          1, HEALTHY)));
+      volumeChecker.setDelegateChecker(
+          new TestStorageVolumeChecker.DummyChecker());
+
+      // 1 for volumeSet and 1 for metadataVolumeSet
```

Comment:
We could remove the old comments here, since we are using ImmutableVolumeSet for test, and the checkAllVolumes call in MutableVolumeSet constructor will not be counted.

Reply:
Thanks for reminding me.  I updated these in a previous attempt, but then started over from scratch.  Now removed.